dask/dataframe/io/parquet/utils.py

```diff
@@ -0,0 +1,305 @@
+            query (cheaply)
+
+        Returns
+        -------
+        meta: pandas.DataFrame
+            An empty DataFrame object to use for metadata.
+            Should have appropriate column names and dtypes but need not have
+            any actual data
+        statistics: Optional[List[Dict]]
+            Either None, if no statistics were found, or a list of dictionaries
+            of statistics data, one dict for every partition (see the next
+            return value).  The statistics should look like the following:
+
+            [
+                {'num-rows': 1000, 'columns': [
+                    {'name': 'id', 'min': 0, 'max': 100, 'null-count': 0},
+                    {'name': 'x', 'min': 0.0, 'max': 1.0, 'null-count': 5},
+                    ]},
+                ...
+            ]
+        parts: List[object]
+            A list of objects to be passed to ``Engine.read_partition``.
+            Each object should represent a row group of data.
+            The type of each object can be anything, as long as the
+            engine's read_partition function knows how to interpret it.
+        """
+        raise NotImplementedError()
+
+    @staticmethod
+    def read_partition(fs, piece, columns, index, **kwargs):
+        """ Read a single piece of a Parquet dataset into a Pandas DataFrame
+
+        This function is called many times in individual tasks
+
+        Parameters
+        ----------
+        fs: FileSystem
+        piece: object
+            This is some token that is returned by Engine.read_metadata.
+            Typically it represents a row group in a Parquet dataset
+        columns: List[str]
+            List of column names to pull out of that row group
+        index: str, List[str], or False
+            The index name(s).
+        **kwargs:
+            Includes `"kwargs"` values stored within the `parts` output
+            of `engine.read_metadata`
+
+        Returns
+        -------
+        A Pandas DataFrame
+        """
+        raise NotImplementedError()
+
+    @staticmethod
+    def initialize_write(df, fs, path, append=False, **kwargs):
+        """Perform engine-specific initialization steps for this dataset
+
+        Parameters
+        ----------
+        df: dask.dataframe.DataFrame
+        fs: FileSystem
+        path: str
+            Output file to write to, usually ``"_metadata"`` in the root of
+            the output dataset; only required if appending.
+        append: bool
+            If True, may use existing metadata (if any) and perform checks
+            against the new data being stored.
+
+        Returns
+        -------
+        tuple:
+            engine-specific instance
+            list of filenames, one per partition
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def write_metadata(parts, meta, fs, path, append=False, **kwargs):
+        """
+        Write a Dask DataFrame to Parquet
```

Comment:
Does this write only the metadata or the full dataframe?  I suspect it's the former, in which case I recommend that we rephrase this docstring header.